src/modules/heads/multi_span_head.py

```diff
@@ -109,6 +108,64 @@ def _get_wordpiece_unified_representation(self, token_representations, wordpiece
 
         #clamped_wordpiece_indices = 
 
+    def _get_logprobs_for_contrastive_training(self, gold_bio_seqs, contrastive_bio_seq, log_probs, mask):
+        """
+        Parameters:
+        -----------
+            gold_bio_seqs - Shape: (B, S, L)
+            contrastive_bio_seq = Shape: (B, L)  single bio-seq per instance
```

Comment:
Are you assuming just one contrastive tag sequence per instance?

Reply:
No, `contrastive_bio_seq` contains multiple [B,I]. Since we want all spans to get low probability they can be packed in the same sequence. This is provided by the reader and read in [L186](https://github.com/nitishgupta/tag-based-multi-span-extraction/pull/1/files/634bd4563a5273085c681a32a92f321122e1b5d0#diff-c57ba957dac25b9cdaf02f6fe166a2f031e0ca8c91cbc2261aa27e99b25f4040R186) 
`gold_bio_seqs` are multiple sequences since you want to marginalize over different combinations of spans. 
Does this seem fine?

Reply:
Ah.. I see. So you're generating just one BIO sequence for multiple spans, and multiple contrastive answers as well. That seems fine, except it is not clear to me how overlapping spans across contrastive answers are handled. Here's an example:
Let's say the sequences for two contrastive answers are
```
OOBIIO
OBIIOO
```
Will the packed sequence be `OBBIIO` or `OBIIIO`? In both options, you seem to be losing out on one of the options in the the denominator for the third position.

Reply:
You're right; I missed this condition. It would be handled however the TASE reader handles such situations. I'll check and update here. 
Consider the union of these possibilities shouldn't be very difficult; do you expect a lot of cases where the contrastive answers would overlap?

Reply:
Okay, thanks! Not sure about DROP, but I think it will be fairly common in Quoref with the first/last/full name differences.

Reply:
Okay, I can try to look at it tomorrow.